README.md

```diff
@@ -13,64 +13,99 @@ pip install . # pip install ProtocolDataUnits
+- **Byte Order Conversion**: Flexibly handle big-endian and little-endian byte orders.
+- **Metadata Storage**: Store additional metadata within PDUs.
+- **Stream Writing/Reading**: Efficiently write and read PDUs to and from streams.
+- **Variable Length Encoding/Decoding**: Handle fields with variable length data.
+- **Pretty Printing of PDUs**: Generate human-readable representations of PDUs.
+- **PDU Equality based on Fields**: Compare PDUs based on their field values.
+- **Serialization and Deserialization**: Serialize PDU definitions to JSON and deserialize them back to PDU objects.
 
 ## Usage
 
+### Creating PDU Formats
+
+You can create PDU formats using the user-friendly API:
+
 ```python
-from ProtocolDataUnits.pdu import PDU, EthernetFrame
-
-# Create a PDU
-pdu = PDU(type=0x01, duration=500, payload=b'\x01\x02\x03\x04')
-encoded_pdu = pdu.encode()
-decoded_pdu = PDU.decode(encoded_pdu)
-
-# Create an Ethernet Frame
-frame = EthernetFrame(
-    dstaddr=(0x01, 0x02, 0x03, 0x04, 0x05, 0x06),
-    srcaddr=(0x11, 0x12, 0x13, 0x14, 0x15, 0x16),
-    ethtype=0x0800,
-    payload=bytes([0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08])
+from ProtocolDataUnits.pdu import create_pdu_format, PDU
+
+# Create a simple PDU format
+my_pdu_format = create_pdu_format(
+    24, 'big',
+    ('uint8', 'type'),
+    ('float', 'value1'),
+    ('double', 'value2')
+)
+
+# Encode data
+encoded_bytes = my_pdu_format.encode({'type': 7, 'value1': 3.14, 'value2': 6.28})
+print(f"Encoded Bytes: {encoded_bytes}")
+
+# Decode data
+decoded_data = my_pdu_format.decode(encoded_bytes)
+print(f"Decoded Data: {decoded_data}")
+```
+
+### Nested PDU Example
+
+```python
+# Define a nested PDU format
+nested_pdu = create_pdu_format(
+    8, 'big',
+    ('uint8', 'nested_type'),
+    ('uint8', 'nested_value')
+)
+
+# Define the main PDU format containing the nested PDU
+main_pdu = create_pdu_format(
+    16, 'big',
+    ('uint8', 'type'),
+    ('nested_pdu', 'nested', nested_pdu)
 )
-encoded_frame = frame.encode()
-decoded_frame = EthernetFrame.decode(encoded_frame)
+
+# Encode data with nested PDU
+encoded_bytes = main_pdu.encode({'type': 7, 'nested': {'nested_type': 1, 'nested_value': 2}})
+print(f"Encoded Bytes: {encoded_bytes}")
+
+# Decode data with nested PDU
+decoded_data = main_pdu.decode(encoded_bytes)
+print(f"Decoded Data: {decoded_data}")
+```
+
+### Serialization and Deserialization
+
+```python
+# Create a PDU with various field types
+my_pdu = PDU().length(68).order('big').uint8('type').float('value1').double('value2').fixed_string('fixed_str', 10).length_prefixed_string('length_str').variable_length_array('array', 'uint8').padding(0xff)
+encoded_bytes = my_pdu.encode({'type': 7, 'value1': 3.14, 'value2': 6.28, 'fixed_str': 'hello', 'length_str': 'dynamic string', 'array': [1, 2, 3, 4, 5]}, compress=True)
+print(f"Encoded Bytes: {encoded_bytes}")
+
+# Decode the PDU
+decoded_data = my_pdu.decode(encoded_bytes, decompress=True)
```

Comment:
Unclear what the `decompress=True` does. Include comment?

Reply:
Added comment for clarity:

```python
# Decode the PDU with decompression enabled.
# decompress=True means that the input data will be decompressed using zlib before decoding.
decoded_data = my_pdu.decode(encoded_bytes, decompress=True)
```